common/include/camera/CCubeProjection.hpp

```diff
@@ -0,0 +1,36 @@
+#ifndef _NBL_CCUBE_PROJECTION_HPP_
+#define _NBL_CCUBE_PROJECTION_HPP_
+
+#include "ILinearProjection.hpp"
+
+namespace nbl::hlsl
+{
+
+template<ProjectionMatrix T>
+struct CCubeProjectionConstraints
+{
+    using matrix_t = typename T;
+    using projection_t = typename IProjection<typename matrix_t>;
+    using projection_range_t = std::array<typename core::smart_refctd_ptr<projection_t>, 6u>;
+    using base_t = ILinearProjection<typename projection_range_t>;
+};
+
+//! Class providing linear cube projections with projection matrix per face of a cube, each projection matrix represents a single view-port
+template<ProjectionMatrix T = float64_t4x4>
```

Comment:
you could actually always make the projection and view matrices 64bit, there's no downsides to that